packages/contracts-bedrock/contracts/test/SafeCall.t.sol

```diff
@@ -39,4 +39,117 @@ contract SafeCall_call_Test is CommonTest {
         assertEq(to.balance, value, "to balance received");
         assertEq(from.balance, 0, "from balance not drained");
     }
+
+    function testFuzz_callWithMinGas_hasEnough_succeeds(
+        address from,
+        address to,
+        uint64 minGas,
+        uint64 value,
+        bytes memory data
+    ) external {
+        vm.assume(from.balance == 0);
+        vm.assume(to.balance == 0);
+        // no precompiles
+        vm.assume(uint160(to) > 10);
+        // don't call the vm
+        vm.assume(to != address(vm));
+        vm.assume(from != address(vm));
+        // don't call the console
+        vm.assume(to != address(0x000000000000000000636F6e736F6c652e6c6f67));
+        // don't call the create2 deployer
+        vm.assume(to != address(0x4e59b44847b379578588920cA78FbF26c0B4956C));
+        // don't send funds to self
+        vm.assume(from != to);
+
+        assertEq(from.balance, 0, "from balance is 0");
+        vm.deal(from, value);
+        assertEq(from.balance, value, "from balance not dealt");
+
+        // Bound minGas to [0, l1_block_gas_limit]
+        minGas = uint64(bound(minGas, 0, 30_000_000));
+
+        vm.expectCallMinGas(to, value, minGas, data);
+        vm.prank(from);
+        bool success = SafeCall.callWithMinGas(to, minGas, value, data);
+
+        assertEq(success, true, "call not successful");
+        assertEq(to.balance, value, "to balance received");
+        assertEq(from.balance, 0, "from balance not drained");
+    }
+
+    function test_callWithMinGas_noLeakageLow_succeeds() external {
+        SimpleSafeCaller caller = new SimpleSafeCaller();
+
+        for (uint64 i = 5000; i < 50000; i++) {
+            uint256 snapshot = vm.snapshot();
+
+            // 26,069 is the exact amount of gas required to make the safe call
+            // successfully.
```

Comment:
I don't love this magic number here that is susceptible to changes in the gas schedule and compiler version, altho i'm not really sure how much better we can do

Reply:
@clabby I think this number is actually important enough to put into Constants.sol

Reply:
hmmm, good question. I need to think about this a bit more...

Reply:
OK, I think it's fine as is.